static/app/views/replays/list/replaySearchBar.tsx

```diff
@@ -63,23 +69,70 @@ function getSupportedTags(supportedTags: TagCollection) {
   };
 }
 
+/**
+ * Returns 3 sections: replay fields, replay click fields, and (filtered, non-field) tags.
+ */
+function getFilterKeySections(
+  supportedTags: TagCollection,
+  organization: Organization
+): FilterKeySection[] {
+  if (!organization.features.includes('search-query-builder-replays')) {
+    return [];
+  }
+
+  const nonFieldTags: string[] = Object.values(supportedTags)
+    .map(tag => tag.key)
+    .filter(
+      key =>
+        !REPLAY_FIELDS.includes(key as ReplayFieldKey | FieldKey) &&
+        !REPLAY_CLICK_FIELDS.includes(key as ReplayClickFieldKey)
+    );
+  nonFieldTags.sort();
+
+  return [
+    {
+      value: 'replay_field',
+      label: t('Replay fields'),
+      children: REPLAY_FIELDS,
+    },
+    {
+      value: 'replay_click_field',
+      label: t('Replay Click Fields'),
+      children: REPLAY_CLICK_FIELDS,
+    },
+    {
+      value: FieldKind.TAG,
+      label: t('Tags'),
+      children: nonFieldTags,
+    },
+  ];
+}
+
 type Props = React.ComponentProps<typeof SmartSearchBar> & {
   organization: Organization;
   pageFilters: PageFilters;
-  placeholder?: string;
 };
 
 function ReplaySearchBar(props: Props) {
-  const {organization, pageFilters, placeholder} = props;
+  const {organization, pageFilters} = props;
   const api = useApi();
   const projectIds = pageFilters.projects;
-  const tags = useTags();
+  const organizationTags = useTags();
```

Comment:
This is how the tags were being fetched before so it doesn't need to be changed in this PR, but I did want to point out that this uses tag data from the global store which fetches them from _all_ datasets. That means that it takes a long time to load and also may contain tags which aren't actually in the replay dataset.

For issues, we are using `useFetchOrganizationTags` and sending the applicable dataset, which returns much faster and is more accurate. Just throwing that out there!